### DISCUSSION TITLE
Binding Alpine directives Directly not working with x-ref

### DISCUSSION THREAD

**Discussion by Drei109** [alpinejs/alpine#1590] [7. Bugs (archived)]
Hi, the x-bind documentation says you can bind any Alpine directives to an element, I tried using it with `x-ref` but I can't make it work. I'm not sure if it's a bug or if I'm doing something wrong.

Here is a codepen showing the issue.
[https://codepen.io/Drei109/pen/mdWYwYJ?editors=1010](https://codepen.io/Drei109/pen/mdWYwYJ?editors=1010)

Thanks by the way for this awesom framework.

**Comment by yabdab** [comment]
The x-ref on myInput1 works, but you do not have one on myInput2  ¯\_(ツ)_/¯ 

On myInput2 add x-ref or x-model ?

**Reply by Drei109** [reply]
I'm adding the `x-ref` directive through `x-bind` in JS. In [this example from the docs](https://alpinejs.dev/directives/bind#bind-directives), they add the directive `x-show` in this way. I thought it should work with `x-ref` too.

``` 
Alpine.data('dropdown', () => ({
            open: false,
            trigger: {
                ['@click']() {
                    this.open = true
                },
            },

            dialogue: {
               // I'm talking about this but with x-ref 👀
                ['x-show']() {
                    return this.open
                },
                ['@click.outside']() {
                    this.open = false
                },
            },
        }))
```

**Reply by yabdab** [reply]
I mainly do most of that inline , in the actual markup, not in data() functions. I am not sure I can help with this one.
